core/utils.lua

```diff
@@ -584,3 +584,43 @@ function format_ui_value(value)
 end
 
 --#endregion
+
+
+function poll_seal(_key, _mod, _guaranteed, _options, _type_key)
+    _mod = _mod or 1
+    local seal_poll = pseudorandom(pseudoseed(_key or 'stdseal'..G.GAME.round_resets.ante))
+    -- If no _options provided, get all seal keys
+    if not _options then
+        _options = get_current_pool("Seal")
+	end
+
+    local available_seals = {}
+    local total_weight = 0
+    for _, v in ipairs(_options) do
+        if v ~= "UNAVAILABLE" then
+            local seal_option = {}
+            if type(v) == 'string' then
+                assert(G.P_SEALS[v])
+                seal_option = { name = v, weight = G.P_SEALS[v].weight or 25 }
```

Comment:
In reference to #155, this should expose the same `get_weight` function that editions do.